scalardb/src/scalardb/transfer.clj

```diff
@@ -26,145 +18,106 @@
-                                                           :before_tx_prepared_at  :bigint
-                                                           :before_tx_committed_at :bigint
-                                                           :primary-key            [:account_id]})))
-  (alia/execute session (alter-table TABLE
-                                     (with {:compaction {:class :SizeTieredCompactionStrategy}}))))
-
-(defn prepare-get
+(def ^:private ^:const KEYSPACE "jepsen")
+(def ^:private ^:const TABLE "transfer")
+(def ^:private ^:const ACCOUNT_ID "account_id")
+(def ^:private ^:const BALANCE "balance")
+
+(def ^:private ^:const INITIAL_BALANCE 10000)
+(def ^:private ^:const NUM_ACCOUNTS 10)
+(def ^:private ^:const TOTAL_BALANCE (* NUM_ACCOUNTS INITIAL_BALANCE))
+
+(def ^:private ^:const SCHEMA {:account_id             :int
+                               :balance                :int
+                               :tx_id                  :text
+                               :tx_version             :int
+                               :tx_state               :int
+                               :tx_prepared_at         :bigint
+                               :tx_committed_at        :bigint
+                               :before_account_id      :int
+                               :before_balance         :int
+                               :before_tx_id           :text
+                               :before_tx_version      :int
+                               :before_tx_state        :int
+                               :before_tx_prepared_at  :bigint
+                               :before_tx_committed_at :bigint
+                               :primary-key            [:account_id]})
+
+(defn- prepare-get
   [id]
   (-> (Key. [(IntValue. ACCOUNT_ID id)])
       (Get.)
       (.forNamespace KEYSPACE)
       (.forTable TABLE)
       (.withConsistency Consistency/LINEARIZABLE)))
 
-(defn prepare-put
-  [id v]
+(defn- prepare-put
+  [id balance]
   (-> (Key. [(IntValue. ACCOUNT_ID id)])
       (Put.)
       (.forNamespace KEYSPACE)
       (.forTable TABLE)
-      (.withValue v)
+      (.withValue (IntValue. BALANCE balance))
       (.withConsistency Consistency/LINEARIZABLE)))
 
-(defn populate-accounts
+(defn- populate-accounts
   "Insert initial records with transaction.
   This method assumes that n is small (< 100)"
   [test n balance]
   (let [tx (scalar/start-transaction test)]
     (try
       (dotimes [i n]
-        (->>
-          (prepare-put i (IntValue. BALANCE balance))
-          (.put tx)))
+        (.put tx (prepare-put i balance)))
       (.commit tx)
       (catch Exception e
         (throw (RuntimeException. (.getMessage e)))))))
 
-(defn calc-new-val
-  "Calculate the new value from the result of transaction.get()"
-  [r amount]
-  (IntValue. BALANCE
-             (-> r .get (.getValue BALANCE) .get .get (+ amount))))
+(defn- get-balance
```

Comment:
Some of the functions are a little duplicate or can be unified in between transfer and transfer-append.
Would it be better to unify ?

Reply:
I can do that for now. But it depends on the column name and the schema.
I don't think it is better to unify them for adding other tests which don't have "balance" column.

Reply:
Yes, it is definitely `transfer` context dependent and not general. So not really sure about what is the best way.